simtools/layout/layout_array_builder.py

```diff
@@ -0,0 +1,244 @@
+            )
+            ax.text(
+                telescopes[i_tel]["pos_x"].value,
+                telescopes[i_tel]["pos_y"].value + telescopes[i_tel]["radius"].value,
+                telescopes[i_tel]["telescope_name"],
+                horizontalalignment="center",
+                verticalalignment="bottom",
+                fontsize=fontsize,
+            )
+
+        plt.gca().add_collection(PatchCollection(patches, match_original=True))
+
+        legend_handler_map = {
+            leg_h.LSTObject: leg_h.LSTHandler(),
+            leg_h.MSTObject: leg_h.MSTHandler(),
+            leg_h.SSTObject: leg_h.SSTHandler(),
+            leg_h.SCTObject: leg_h.SCTHandler(),
+        }
+
+        x_title = "East [m]"
+        y_title = "North [m]"
+        plt.axis("square")
+        plt.grid(True)
+        plt.gca().set_axisbelow(True)
+        plt.xlabel(x_title, fontsize=18, labelpad=0)
+        plt.ylabel(y_title, fontsize=18, labelpad=0)
+        plt.tick_params(axis="both", which="major", labelsize=15)
+
+        plt.legend(
+            legend_objects,
+            legend_labels,
+            handler_map=legend_handler_map,
+            prop={"size": 11},
+            loc="best",
+        )
+
+        plt.tight_layout()
+
+        return fig
+
+    @staticmethod
+    def telescope_layout_file_to_dict(file_name):
+        """
+        Read the telescope layout from ECSV layout file and returns a dictionary with the x, y, z \
+        positions and the radius.
+        Note: file_name is usually generated by self.export_telescope_list.
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the ecsv file with telescope layout.
+
+        Returns
+        -------
+        dict
+            Dictionary with the telescope layout information.
+        """
+
+        telescopes = QTable.read(file_name)
+        telescopes["radius"] = [
+            float(telescopes.meta["corsika_sphere_radius"][tel_name_now[:3]].split()[0])
+            for tel_name_now in telescopes["telescope_name"]
+        ]
+        telescopes["radius"].unit = u.Unit(
+            telescopes.meta["corsika_sphere_radius"]["LST"].split()[1]
+        )
+
+        return telescopes
+
+    @u.quantity_input(x=u.m, y=u.m, radius=u.m)
+    def _get_telescope_patch(self, name, x, y, radius):
+        """
+        Collect the patch of one telescope to be plotted by self.plot_array.
+
+        Parameters
+        ----------
+        name: str
+            Name of the telescope (type).
+        x: astropy.units.m
+            X position of the telescope in meters.
+        y: astropy.units.m
+            Y position of the telescope in meters.
+        radius: astropy.units.m
+            Radius of the telescope sphere in meters.
+
+        Returns
+        -------
+        patch
+            Instance of mpatches.Circle.
+        """
+
+        valid_name = names.get_telescope_type(name)
+        fill_flag = False
+        if valid_name == "MST":
```

Comment:
Hardcoded telescopes names. Wonder if there is another solution; does the introduction of a new telescope type in CTA means that we have to update this code? 

There is also MST and SCT here, why no SST or LST?

Reply:
I use now names.all_telescope_types for accessing the names but individual types have to be specifically written. we intentionally chose this plot marker for the MST, as we decided also for a specific marker for the SCT. We dont expect telescope types to change very often but definitely in case CTA eventually decides for another type of telescope, we would have to include it in the code as well.

The other telescopes will get the fill_flag false as defined before and the Circle as a patch as given by the else statement in the sequence.

Reply:
But we haven't even agreed (in CTA) how telescopes are named. If we hardwire names now in many places, it will be very hard to change later on. I propose to have telescope names in one single place only.